libp2p/transports/tls/certificate.c

```diff
@@ -959,4 +959,87 @@ void cert_free_key(cert_key_t key) {
   struct cert_key_s *k = (struct cert_key_s *)key;
   EVP_PKEY_free(k->pkey);
   free(k);
+}
+
+
+cert_error_t cert_signing_req(const char *cn, cert_key_t key, cert_buffer **csr_buffer) {
+  cert_error_t ret_code = CERT_SUCCESS;
+  X509_REQ *x509_req = NULL;
+  X509_NAME *name = NULL;
+  X509_EXTENSION *ext = NULL;
+  X509V3_CTX ctx;
+  STACK_OF(X509_EXTENSION) *exts = NULL;
+  unsigned char *der = NULL;
+  size_t der_len = 0;
+
+  if (!key || !(key->pkey)) {
+    ret_code = CERT_ERROR_NO_PUBKEY;
+    goto cleanup;
+  }
+  EVP_PKEY *pkey = key->pkey;
+
+  x509_req = X509_REQ_new();
+  if (!x509_req) {
+    ret_code = CERT_ERROR_X509_REQ_GEN;
+    goto cleanup;
+  }
+
```

Comment:
Quickly reading the RFC 6125, even tho the CN is not required **for validation**, it still should exist. Interestingly enough, you can just truncate the content for cases in which the cn exceeds 64 bytes.

Maybe we can  add something like this?
```cpp
char truncated_cn[65]; // 64 + null terminator
strncpy(truncated_cn, cn, 64);
truncated_cn[64] = '\0'; // ensure null termination

if (!X509_NAME_add_entry_by_txt(name, "CN", MBSTRING_ASC,
        (unsigned char *)truncated_cn, -1, -1, 0)) {
  ret_code = CERT_ERROR_CN;
  goto cleanup;
}
```

Reply:
I found that Let's Encrypt misbehaves when CN is set.